Gems/Prefab/PrefabDependencyViewer/Code/Source/MainWindow.h

```diff
@@ -0,0 +1,139 @@
+#include <GraphCanvas/Widgets/GraphCanvasEditor/GraphCanvasAssetEditorMainWindow.h>
+#include <GraphCanvas/Widgets/GraphCanvasEditor/GraphCanvasEditorDockWidget.h>
+#include <GraphCanvas/Widgets/NodePalette/NodePaletteDockWidget.h>
+#include <QWidget>
+#include <MainWindowInterface.h>
+#include <AzCore/std/containers/unordered_map.h>
+
+namespace PrefabDependencyViewer
+{
+    struct PrefabDependencyViewerConfig
+        : GraphCanvas::AssetEditorWindowConfig
+    {
+        /** Return an empty NodePalette tree */
+        GraphCanvas::GraphCanvasTreeItem* CreateNodePaletteRoot() override;
+    };
+
+    using Endpoint = GraphCanvas::Endpoint;
+    using ConnectionId = GraphCanvas::ConnectionId;
+
+    class PrefabDependencyViewerWidget 
+        : public GraphCanvas::AssetEditorMainWindow
+        , public PrefabDependencyViewerInterface
+        , private GraphCanvas::GraphModelRequestBus::Handler
+    {
+        Q_OBJECT;
+    public:
+        explicit PrefabDependencyViewerWidget(QWidget* pParent = NULL);
+        virtual ~PrefabDependencyViewerWidget();
+
+        ////////////////// GraphCanvas::AssetEditorMainWindow overrides //////////////
+        /** Sets up the GraphCanvas UI without the Node Palette. */
+        virtual void SetupUI() override;
+
+        ////////////////// PrefabDependencyViewerWidget overrides ////////////////////
+        virtual void DisplayTree(const Utils::DirectedGraph& graph) override;
+
+        void DisplayNodesByLevel(const Utils::DirectedGraph& graph, AZStd::vector<int> numNodesAtEachLevel, int widestLevel);
+
+        void DisplayNode(Utils::Node* node, AZ::Vector2 pos);
+
+        
+        GraphCanvas::SlotId CreateExecutionSlot(
+            GraphCanvas::NodeId nodeId,
+            const AZStd::string& slotName,
+            const AZStd::string& tooltip,
+            GraphCanvas::SlotGroup slotGroup,
+            bool isInput);
+
+        GraphCanvas::SlotId CreateDataSlot(
+            GraphCanvas::NodeId nodeId,
+            const AZStd::string& slotName,
+            const AZStd::string& tooltip,
+            AZ::Uuid dataType,
+            GraphCanvas::SlotGroup slotGroup,
+            bool isInput);
+
+        void AddSlotToNode(AZ::Entity* slotEntity, GraphCanvas::NodeId nodeId);
+        void CreateNodeUi(const AzToolsFramework::Prefab::TemplateId& tid);
+
+    protected:
+        ////////////////// GraphCanvas::AssetEditorMainWindow overrides //////////////
+        /** Overriding RefreshMenu in order to remove the
+        unnecessary menu bar on the top. As a bonus, this
+        also removes the ability to revive NodePalette from the UI. */
+        virtual void RefreshMenu() override {}
+
+        // GraphCanvas::GraphModelRequestBus::Handler overrides
+        virtual void RequestUndoPoint() {}
+        virtual void RequestPushPreventUndoStateUpdate() {}
+        virtual void RequestPopPreventUndoStateUpdate() {}
+        virtual void TriggerUndo() {}
+        virtual void TriggerRedo() {}
```

Comment:
Shouldn't these (and possibly a few functions below as well) be tagged with `override`?

Reply:
Resolved